libs/s25main/WindowManager.cpp

```diff
@@ -876,3 +876,59 @@ void WindowManager::DrawToolTip()
         curTooltip->draw(ttPos);
     }
 }
+
+DrawPoint WindowManager::SnapWindow(Window* wnd, const Rect& wndRect) const
+{
+    const auto snapDistance = static_cast<int>(SETTINGS.interface.windowSnapDistance);
+    if(snapDistance == 0)
+        return DrawPoint::all(0); // No snapping
+
+    /// The smallest distance to another window edge; values greater than snapDistance indicate no snapping
+    auto minDist = Extent::all(snapDistance + 1);
+    /// The smallest offset (i.e., signed distance) to another window edge; zero indicates no snapping
+    DrawPoint minOffset = DrawPoint::all(0);
+
+    auto setOffsetIfLowerDist = [](int a, int b, unsigned& minDist, int& minOffset) {
+        const int offset = b - a;
+        const unsigned dist = std::abs(offset);
+        if(dist < minDist)
+        {
+            minDist = dist;
+            minOffset = offset;
+        }
+    };
+
+    for(const auto& curWnd : windows)
+    {
+        if(curWnd.get() == wnd)
+            continue;
+
+        const auto setOffsetIfLowerDistX = [&](int x1, int x2) {
+            setOffsetIfLowerDist(x1, x2, minDist.x, minOffset.x);
+        };
+        const auto setOffsetIfLowerDistY = [&](int y1, int y2) {
+            setOffsetIfLowerDist(y1, y2, minDist.y, minOffset.y);
+        };
```

Comment:
Those lambdas can now be moved outside the loop can't they?
```suggestion
    const auto setOffsetIfLowerDistX = [&](int x1, int x2) {
        setOffsetIfLowerDist(x1, x2, minDist.x, minOffset.x);
    };
    const auto setOffsetIfLowerDistY = [&](int y1, int y2) {
        setOffsetIfLowerDist(y1, y2, minDist.y, minOffset.y);
    };

    for(const auto& curWnd : windows)
    {
        if(curWnd.get() == wnd)
            continue;

```